### PR TITLE
Adding --min-trades and --max-trades for hyperopt-list

### PR DESCRIPTION
**Pull request by Fredrik81**
## Summary
Adding more filter options for hyperopt-list to easier find what your looking for in larger runs

PEP8 = Yes
## Quick changelog
Adding two more filter options that can be used in combination to the other options in order to filter and search thru larger hyperopt runs.

- Adding filter options --max-trades and --min-trades
- Adding missing filter options --max-avg-profit and --max-total-profit
- Added additional tests for the new options

## What's new?
```
freqtrade hyperopt-list --max-avg-time 200 --min-trades 20 --max-avg-profit 0.29 --min-avg-profit 0.20 --max-total-profit 8
2020-02-11 21:01:47,383 - freqtrade.loggers - INFO - Verbosity set to 0
2020-02-11 21:01:47,383 - freqtrade.configuration.configuration - INFO - Parameter --min-trades detected: 20
2020-02-11 21:01:47,383 - freqtrade.configuration.configuration - INFO - Parameter --max-avg-time detected: 200.0
2020-02-11 21:01:47,383 - freqtrade.configuration.configuration - INFO - Parameter --min-avg-profit detected: 0.2
2020-02-11 21:01:47,383 - freqtrade.configuration.configuration - INFO - Parameter --max-avg-profit detected: 0.29
2020-02-11 21:01:47,383 - freqtrade.configuration.configuration - INFO - Parameter --max-total-profit detected: 8.0
2020-02-11 21:01:47,383 - freqtrade.configuration.check_exchange - INFO - Checking exchange...
2020-02-11 21:01:47,383 - freqtrade.configuration.configuration - INFO - Using pairlist from configuration.
2020-02-11 21:01:47,383 - freqtrade.configuration.config_validation - INFO - Validating configuration ...
2020-02-11 21:01:47,574 - freqtrade.optimize.hyperopt - INFO - Loaded 1096 previous evaluations from disk.
2020-02-11 21:01:47,574 - freqtrade.commands.hyperopt_commands - INFO - 1 epochs found.
   602/1096:     29 trades. Avg profit   0.27%. Total profit  0.00046184 BTC (   7.69Σ%). Avg duration 189.8 min. Objective: 2.07836

Best result:

   602/1096:     29 trades. Avg profit   0.27%. Total profit  0.00046184 BTC (   7.69Σ%). Avg duration 189.8 min. Objective: 2.07836
```

### CODE REVIEW

File: freqtrade/commands/cli_options.py [hunk 1]
```diff
@@ -398,6 +398,18 @@ def __init__(self, *args, **kwargs):
         help='Select only best epochs.',
         action='store_true',
     ),
+    "hyperopt_list_min_trades": Arg(
+        '--min-trades',
+        help='Select epochs with more than INT trades.',
+        type=check_int_nonzero,
```
Comment (hroff-1902) [inline freqtrade/commands/cli_options.py]: better use `check_int_positive` here?

File: freqtrade/commands/cli_options.py [hunk 2]
```diff
@@ -398,6 +398,18 @@ def __init__(self, *args, **kwargs):
...
+        metavar='INT',
+    ),
+    "hyperopt_list_max_trades": Arg(
+        '--max-trades',
+        help='Select epochs with less than INT trades.',
+        type=check_int_nonzero,
```
Comment (hroff-1902) [inline freqtrade/commands/cli_options.py]: same

Comment (Fredrik81) [reply freqtrade/commands/cli_options.py]: your correct. will change

File: freqtrade/commands/cli_options.py (hunk 1 above)
Comment (Fredrik81) [reply freqtrade/commands/cli_options.py]: your correct. will change

**Comment by Fredrik81** [comment]
will add a couple of more to get completeness in the PR. Wait with review

**Comment by hroff-1902** [comment]
Finally LGTM 👍 , thanks!

**Pull request merged**